unit-tests/src/test/scala/java/util/CollectionTest.scala

```diff
@@ -0,0 +1,296 @@
+    assertTrue(coll.remove("one"))
+    assertEquals(initialSize - 2, coll.size())
+  }
+
+  @Test def shouldRemoveStoredElementsOnDoubleCornerCases(): Unit = {
+    val coll = factory.empty[Double]
+
+    coll.add(1.234)
+    coll.add(2.345)
+    coll.add(Double.NaN)
+    coll.add(+0.0)
+    coll.add(-0.0)
+
+    // coll == ArrayCollection(1.234, 2.345, NaN, +0.0, -0.0)
+    assertTrue(coll.remove(Double.NaN))
+    // coll == ArrayCollection(1.234, 2.345, +0.0, -0.0)
+    assertEquals(4, coll.size())
+    assertTrue(coll.remove(2.345))
+    // coll == ArrayCollection(1.234, +0.0, -0.0)
+    assertEquals(3, coll.size())
+    assertTrue(coll.remove(1.234))
+    // coll == ArrayCollection(+0.0, -0.0)
+    assertEquals(2, coll.size())
+    assertTrue(coll.remove(-0.0))
+    // coll == ArrayCollection(NaN, +0.0)
+    assertEquals(1, coll.size())
+
+    coll.clear()
+
+    assertTrue(coll.isEmpty)
+  }
+
+  @Test def shouldBeClearedWithOneOperation(): Unit = {
+    val coll = factory.empty[String]
+
+    coll.add("one")
+    coll.add("two")
+    assertEquals(2, coll.size)
+    coll.clear()
+    assertEquals(0, coll.size)
+  }
+
+  @Test def shouldCheckContainedPresence(): Unit = {
+    val coll = factory.empty[String]
+
+    coll.add("one")
+    assertTrue(coll.contains("one"))
+    assertFalse(coll.contains("two"))
+    if (factory.allowsNullElementQuery) {
+      assertFalse(coll.contains(null))
+    } else {
+      expectThrows(classOf[Exception], coll.contains(null))
+    }
+  }
+
+  @Test def shouldCheckContainedPresenceForDoubleCornerCases(): Unit = {
+    val coll = factory.empty[Double]
+
+    coll.add(-0.0)
+    assertTrue(coll.contains(-0.0))
+    assertFalse(coll.contains(+0.0))
+
+    coll.clear()
+
+    coll.add(+0.0)
+    assertFalse(coll.contains(-0.0))
+    assertTrue(coll.contains(+0.0))
+  }
+
+  @Test def shouldGiveProperIteratorOverElements(): Unit = {
+    val coll = factory.empty[String]
+    coll.add("one")
+    coll.add("two")
+    coll.add("three")
+    coll.add("three")
+    coll.add("three")
+
+    assertIteratorSameElementsAsSetDupesAllowed("one", "two", "three")(
+      coll.iterator())
+  }
+
+  // Issue: https://github.com/scala-native/scala-native/issues/1972
+  @Ignore(
+    "removeIf is a JavaDefaultMethod which is not supported yet on Scala 2.11")
+  @Test def removeIf(): Unit = {
+    // val coll = factory.fromElements[Int](42, 50, 12, 0, -45, 102, 32, 75)
+    // assertEquals(8, coll.size())
+
+    // assertTrue(coll.removeIf(new java.util.function.Predicate[Int] {
+    //   def test(x: Int): Boolean = x >= 50
+    // }))
+    // assertEquals(5, coll.size())
+    // assertIteratorSameElementsAsSet(-45, 0, 12, 32, 42)(coll.iterator())
+
+    // assertFalse(coll.removeIf(new java.util.function.Predicate[Int] {
+    //   def test(x: Int): Boolean = x >= 45
+    // }))
+    // assertEquals(5, coll.size())
+    // assertIteratorSameElementsAsSet(-45, 0, 12, 32, 42)(coll.iterator())
+  }
```

Comment:
I think you can now uncomment this test and remove the `// Issue: ...` comment I added and the `@Ignore` annotation. Take [the original file](https://github.com/scala-js/scala-js/blob/4eac249d0e4a922f5fcdb7c8e7c48c95260438c0/test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionTest.scala) as guide.

Reply:
Oops!  Thank you for catching that.  It is kind of one of the main motivators for the Test.
I had read your documentation about the `ignore` and seen the code, but the message
never arrived to my fingers when I copied the code.  Fix in process.